backpack/extensions/secondorder/diag_ggn/losses.py

```diff
@@ -7,17 +7,22 @@
 
 class DiagGGNLoss(DiagGGNBaseModule):
     def backpropagate(self, ext, module, grad_inp, grad_out, backproped):
+        hess_func = self.make_loss_hessian_func(ext)
 
-        if ext.loss_hessian_strategy == LossHessianStrategy.EXACT:
-            hess_func = self.derivatives.sqrt_hessian
-        elif ext.loss_hessian_strategy == LossHessianStrategy.SAMPLING:
-            hess_func = self.derivatives.sqrt_hessian_sampled
+        return hess_func(module, grad_inp, grad_out)
+
+    def make_loss_hessian_func(self, ext):
+        """Get function that produces the backpropagated quantity."""
+        loss_hessian_strategy = ext.loss_hessian_strategy
+
+        if loss_hessian_strategy == LossHessianStrategy.EXACT:
+            return self.derivatives.sqrt_hessian
+        elif loss_hessian_strategy == LossHessianStrategy.SAMPLING:
+            mc_samples = ext.get_num_mc_samples()
+            return self.derivatives.make_sqrt_hessian_sampled_fn(mc_samples)
```

Comment:
You can also make the partial evaluation here directly, rather than use `make_sqrt_hessian_sampled_fn`.
```
return partial(self.sqrt_hessian_sampled, mc_samples=mc_samples)
```